src/components/interactive/dragBoxLayer.ts

```diff
@@ -0,0 +1,327 @@
+///<reference path="../../reference.ts" />
+
+module Plottable {
+export module Component {
+  export module Interactive {
+    type _EdgeIndicator = {
+      top: boolean;
+      bottom: boolean;
+      left: boolean;
+      right: boolean;
+    }
+
+    export class DragBoxLayer extends Component.SelectionBoxLayer {
+      private _dragInteraction: Interaction.Drag;
+      private _detectionEdgeT: D3.Selection;
+      private _detectionEdgeB: D3.Selection;
+      private _detectionEdgeL: D3.Selection;
+      private _detectionEdgeR: D3.Selection;
+      private _detectionCornerTL: D3.Selection;
+      private _detectionCornerTR: D3.Selection;
+      private _detectionCornerBL: D3.Selection;
+      private _detectionCornerBR: D3.Selection;
```

Comment:
? We removed "box" previously in favor of this though. Since it's a private var I don't think it matters that much...